File: Data Structures & Algorithms/two-integer-sum/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numbers = new HashMap<>(); //create hashmap
        int small = 0; //initialize smaller
        int large = 0; //initialize larger
        for (int i = 0; i < nums.length; i++){
            int diff = target - nums[i]; //checks hashmap for other term 
            if (numbers.containsKey(diff)){ //if has it update small + large
                small = Math.min (numbers.get(diff),i);
                large = Math.max (numbers.get(diff),i);
            } else {
                numbers.put(nums[i], i); //if doesn't we add to hashmap
            }
        }
        int [] indices = {small,large}; //initialize return array
        return indices;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-2.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        
        l,r = 0, len(heights)-1
        most_water = 0

        while l<r:
            current_water = min(heights[l], heights[r]) * (r-l)
            most_water = max(current_water, most_water)

            if heights[l] < heights[r]:
                l += 1
            else:
                r -= 1
        
        return most_water
